plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,190 +82,414 @@ function executeCLI($scriptName, &$args) {
+
+		if (count($missingHeaders)) {
+			echo __('pugin.importexport.csv.missingHeadersOnCsv', ['missingHeaders' => $missingHeaders]);
+			exit(1);
+		}
+
+		// Imports
+		import('lib.pkp.classes.submission.SubmissionFile'); // constants.
+		import('lib.pkp.classes.file.FileManager');
+		import('lib.pkp.classes.core.Core');
+		import('classes.file.PublicFileManager');
+
+		// DAOs
+		$pressDao = Application::getContextDAO();
+		$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
+		$publicationDao = DAORegistry::getDAO('PublicationDAO'); /* @var $publicationDao PublicationDAO */
+		$seriesDao = DAORegistry::getDAO('SeriesDAO'); /* @var $seriesDao SeriesDAO */
+		$authorDao = DAORegistry::getDAO('AuthorDAO'); /* @var $authorDao AuthorDAO */
+		$publicationFormatDao = DAORegistry::getDAO('PublicationFormatDAO'); /* @var $publicationFormatDao PublicationFormatDAO */
+		$publicationDateDao = DAORegistry::getDAO('PublicationDateDAO'); /* @var $publicationDateDao PublicationDateDAO */
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
+		$categoryDao = DAORegistry::getDAO('CategoryDAO'); /* @var $categoryDao CategoryDAO */
+
+		// Static variables
+		$fileManager = new FileManager();
+		$publicFileManager = new PublicFileManager();
+		$dirNames = Application::getFileDirectories();
+		$format = trim($dirNames['context'], '/') . '/%d/' . trim($dirNames['submission'], '/') . '/%d';
+		$allowedFileTypes = ['gif', 'jpg', 'png', 'webp'];
+
+		/** @var \PKP\services\PKPFileService */
+		$fileService = Services::get('file');
+		/** @var \PKP\services\PKPPublicationService */
+		$publicationService = Services::get('publication');
+
+		// Cache variables
+		$genres = [];
+		$authorGroups = [];
+		$series = [];
+		$cachedCategories = [];
+		$presses = [];
+
+		$csvForInvalidRowsName = "{$basePath}/invalid_rows.csv";
+		$invalidRowsFile = new SplFileObject($csvForInvalidRowsName, 'w');
+
+		$invalidRowsFile->fputcsv(array_merge(array_pad($headers, 15, null), ['error']));
+
+		foreach ($file as $index => $fields) {
+			if (!$index) {
+				continue; // Skip headers
+			}
+
+			if (empty(array_filter($fields))) {
+				continue;
+			}
+
+			// Verifica se chegou ao final do conteúdo
+			if (count($fields) < 15) {
+				$invalidRowsFile->fputcsv(array_merge(array_pad($fields, 15, null), [__('plugins.importexport.csv.csvDoesntContainAllFields')]));
+				continue;
+			}
+
+			$data = (object) array_combine($headers, array_pad(array_map('trim', $fields), count($headers), null));
+			if (!$this->requiredFieldsPresent($data)) {
+				$invalidRowsFile->fputcsv(array_merge(array_pad($fields, 15, null), [__('plugins.importexport.csv.requiredFieldsMissing')]));
```

Comment:
You can replace all occurrences of 15 by `count($expectedHeaders)`, it can be stored on a variable.